config/services.yaml

```diff
@@ -17,6 +17,12 @@ services:
         resource: '../src/*'
         exclude: '../src/{DependencyInjection,Entity,Migrations,Tests,Kernel.php}'
 
+    App\Analyser\FileCache:
+        $cacheFile: '%cache.file%'
+
+    App\Analyser\MemoizingAnalyser:
+        $analyser: '@App\Analyser\RstAnalyser'
```

Comment:
```suggestion
        App\Analyser\Analyser $analyser: '@App\Analyser\RstAnalyser'
```

Reply:
This does not work

Reply:
why?

Reply:
Isn't only allowed in `bind`? -> https://symfony.com/doc/current/service_container.html#binding-arguments-by-name-or-type

As I got following error:
```
The configuration key "string $cacheFile" is unsupported for definition "App\Analyser\FileCache" in "/Users/simon/Projects/doctor-rst/config//services.yaml".
```

Reply:
Oh you are right, so this is working without `bind`?

This would work?
```yaml
    App\Analyser\MemoizingAnalyser:
        bind:
            My\FunkyInterface $analyser: '@App\Analyser\RstAnalyser'
```
??

Reply:
It isn't worth it. PHP checks the type.